doc/build/changelog/migration_10.rst

```diff
@@ -167,7 +167,7 @@ applied::
         @declared_attr.cascading
         def id(cls):
             if has_inherited_table(cls):
-                return Column(ForeignKey('myclass.id'), primary_key=True)
+                return Column('some_id', ForeignKey('myclass.id'))
             else:
```

Comment:
Details of change see my [comment](https://github.com/sqlalchemy/sqlalchemy/pull/5371/files#r436363759)

Reply:
Column `id` will be created in my version of the example. And more precisely, the `id` will be inherited from the parent class.  Note: I suppose that this example of  "Single Table Inheritance". Result SQL:
```sqL
CREATE TABLE myclass (
	id INTEGER NOT NULL, 
	some_id INTEGER, 
	PRIMARY KEY (id), 
	FOREIGN KEY(some_id) REFERENCES myclass (id)
)
```
Additional we can look at init [commit](https://github.com/sqlalchemy/sqlalchemy/blame/079df65dc0f71ea4d1771b6ae17e13242c766517/lib/sqlalchemy/ext/declarative/api.py#L196-L220) that added this example:

>         Below, both MyClass as well as MySubClass will have a distinct
>        ``id`` Column object established::
>         ...
>        The behavior of the above configuration is that ``MySubClass``
>        will refer to both its own ``id`` column as well as that of
>        ``MyClass`` underneath the attribute named ``some_id``.